### PR TITLE
Adding option and logic to optionally skip folder creation

### LINKED ISSUE #4087: [ACTION] Add functionality to Google Drive > Create Folder to only create if the folder doesn't exist
Google Drive allows duplicate folder names, but this is not always what we want. When creating a folder, it would be nice to have the option to ONLY create if the folder doesn't yet exist. 

**Please provide a link to the relevant API docs for the specific service / operation.**
Google's Drive API doesn't support this option natively, so it would be necessary to do a quick search before creating the folder. Here's the doc page for `files.list`: https://developers.google.com/drive/api/guides/search-files

### PR DESCRIPTION
**Pull request by bsommardahl**
fixes #4087

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4088"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thank you so much for submitting this! We've added it to our backlog to review, and our team has been notified.

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

**Comment by feyzullah** [comment]
Hi @bsommardahl can you please increment action version `0.0.4` => `0.0.5`

**Review by feyzullah — APPROVED** [review]

**Comment by vunguyenhung** [comment]
Hello @bsommardahl, thank you so much for you contribution 🙏 

I've attempted to test this PR and there're the following issues:

1. Error when publishing action `{"error":"{\"name\":\"UserError\",\"message\":\"ReferenceError: toSingleLineString is not defined\"}"}`image
2. Lint errors still present (you can check lint error by cloning the pipedream project, execute `pnpm i --recursive` then execute `npx eslint components/google_drive/actions/create-folder/create-folder.mjs`) Screen Shot 2022-08-25 at 23 46 43

Could you fix them and ping us to review again?

**Comment by feyzullah** [comment]
Hi @bsommardahl, @vunguyenhung this is my bad, I forgot to run `eslint`. Fixed linter errors and modified the action because we were not able to detect if the folder name was included in another folder name. For example if we have a folder named `test-it` and creating another folder named `test`, it was not able to differentiate them and was not creating another folder. With this update, it should have been fixed.

**Review by feyzullah — APPROVED** [review]

**Comment by bsommardahl** [comment]
Thanks for fixing that! I forgot to import the function, so that was my bad.

File: components/google_drive/actions/create-folder/create-folder.mjs [hunk 1]
```diff
@@ -36,12 +42,39 @@ export default {
...
+      }))).files;
+      for (let f of folders) {
+        if (f.name == name) {
+          folder = f;
+          break;
+        }
+      }
+      if (folder) {
+        $.export("$summary", "Found existing folder, therefore not creating folder. Returning found folder.");
+        return folder;
```
Comment (vunguyenhung) [inline components/google_drive/actions/create-folder/create-folder.mjs:75]:
```suggestion
        $.export("$summary", "Found existing folder, therefore not creating folder. Returning found folder.");
        const folderDetails = await this.googleDrive.getFile(folder.id);

        return folderDetails;
```

**Comment by vunguyenhung** [comment]
Hi everyone, I have added some improvement commits and confirm that all test cases are passed with those changes! Ready for release!

Test report
[Google_Drive_4088.pdf](https://github.com/PipedreamHQ/pipedream/files/9432019/Google_Drive_4088.pdf)

@feyzullah would you mind reviewing my commit the last time before merging?

**Review by feyzullah — APPROVED** [review]

**Pull request merged**